### DISCUSSION TITLE
Is there a way to prevent compromised collaborator accounts from stealing secrets?

### DISCUSSION THREAD

**Discussion by jazonxyz** [community/community#25047] [Actions]
<p>If I understand correctly, a single collaborator account could be compromised and a malicious actor could modify a config to run when pushed and send all your secrets to their server.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Sorry to tell you that you could not prevent collaborators to access to secrets through workflow.  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0c46b4386bd71b8c73cfd54c36d9ef9f6448d5e7.png" data-download-href="/uploads/short-url/1KBd9PVc2tmb56lJFoM4tgkTUCb.png?dl=1" title="secrets.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088552-1928ebff-86ce-4f66-a518-ae359f8c63a3.png" alt="secrets.png" data-base62-sha1="1KBd9PVc2tmb56lJFoM4tgkTUCb" width="690" height="467" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0c46b4386bd71b8c73cfd54c36d9ef9f6448d5e7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">secrets.png</span><span class="informations">715×484 17.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jazonxyz** [comment]
<p>Oh bummer.  I was hoping for a way to secure these for when working with contractors.  Thanks for the reply!</p>
